internal/core/sapsystem/sapsystem.go

```diff
@@ -80,7 +83,7 @@ func NewSAPSystemsList(
 	var systems = SAPSystemsList{}
```

Comment:
I know it's old code but we want to refactort this to the default/new constructor convention we have in the other gatherers? To inject dependencies in tests and the default one with the real scenario?

Reply:
I get your point (even though this is not a gatherer code).
Refactoring a bit the code would allow us to test better (in fact, this particular function is not even tested...).
Do you mind if I open a tech-debt ticket in our backlog?
The refactor might not be so trivial, and I don't want to include it in this PR